### PR TITLE
Ros CPP Bag recorder

### PR DESCRIPTION
**Pull request by franklinv77**

### CODE REVIEW

File: examples/ros-bag-recorder-cpp/readme.txt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+
+
+To build proto files:
+
+    cd src/protos
+    sudo protoc -I . --grpc_out=. --plugin=protoc-gen-grpc=`which grpc_cpp_plugin` protos/*/v1/*.proto
+    sudo protoc -I . --cpp_out=. protos/*/v1/*.proto
+
+
+sudo apt install ros-noetic-ros-type-introspection
```
Comment (AaronGBoswell) [inline examples/ros-bag-recorder-cpp/readme.txt:10]: is it normal to add 'someone else's protos' to the project like this? what if we checked out the protos from https://github.com/FormantIO/formant/tree/master/protos as part of the build step, and added it to the gitignore

File: examples/ros-bag-recorder-cpp/src/adapter/package.xml [hunk 2]
```diff
@@ -0,0 +1,62 @@
+<?xml version="1.0"?>
```
Comment (AaronGBoswell) [inline examples/ros-bag-recorder-cpp/src/adapter/package.xml:1]: seems like this file needs a lot of work

File: examples/ros-bag-recorder-cpp/readme.txt (hunk 1 above)
Comment (franklinv77) [reply examples/ros-bag-recorder-cpp/readme.txt:10]: I think that is probably a better way to do this

File: examples/ros-bag-recorder-cpp/src/adapter/src/NOTES.txt [hunk 3]
```diff
@@ -0,0 +1,3 @@
+
+To get the ros introspection lib:
+    sudo apt install ros-noetic-ros-type-introspection
```
Comment (AaronGBoswell) [inline examples/ros-bag-recorder-cpp/src/adapter/src/NOTES.txt:3]: what if they're not noetic? `ros-$ROS_DISTRO-ros-type-instrospection` maybe ?

File: examples/ros-bag-recorder-cpp/src/adapter/src/baghandler.hpp [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+        bag2_end_time = bag2_start_time + ros::Duration(2 * bag_length - bag_overlap);
+    }
+
+    inline ~BagHandler(){
+        bag1.close();
+        bag2.close(); 
+    }
+
+    /**
+     * @brief Write the outgoing message to the current open rosbag's.
```
Comment (AaronGBoswell) [inline examples/ros-bag-recorder-cpp/src/adapter/src/baghandler.hpp:90]: not posessive (')

File: examples/ros-bag-recorder-cpp/src/adapter/src/rsjp.hpp [hunk 5]
```diff
@@ -0,0 +1,785 @@
+/** **************************************************************************************
```
Comment (AaronGBoswell) [inline examples/ros-bag-recorder-cpp/src/adapter/src/rsjp.hpp:1]: is it normal to include this file in your project directly?

**Review by AaronGBoswell — APPROVED** [review]

**Pull request merged**